pkg/apis/maistra/conversion/conversion_v1_to_v2.go

```diff
@@ -230,6 +230,13 @@ func Convert_v1_ControlPlaneSpec_To_v2_ControlPlaneSpec(in *v1.ControlPlaneSpec,
 		}
 	}
 
+	// Discovery Selectors
+	if version.AtLeast(versions.V2_4) {
+		if err := populateDiscoverySelectorsConfig(values, out); err != nil {
+			return err
+		}
+	}
```

Comment:
You can merge this if statement with the one for extension providers. No need to have two if statements that check the version.